### DISCUSSION TITLE
How to add a payload to the packet?

### DISCUSSION THREAD

**Discussion by orangepineapple** [ANLAB-KAIST/KENSv3#222] [Project Q&A 2. Transport Layer (TCP)] [answered]
I tried to add a payload to the TCP packet by just increasing the packet size. But then when I tried to test this against PA2 test cases it failed every test case. I can't figure out why this is happening when I add an empty payload.
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/57548877/6115e333-7f9f-4d7f-8280-4dd5b318a84b)

**Comment by Pusnow** [comment]
How do you resize the packets?

For now, you can create a new packet and copy original packet's contents to new one.
I admit that it is little inconvenient to resize packets. I'll add an easy method for resizing soon.

**Reply by orangepineapple** [reply]
I just changed the input size to the packet constructor, and then kept the same code for writing the header into the packet

**Reply by Pusnow** [reply, marked as answer]
Did you calculate checksum correctly?

Try turning on checksum verification in Wireshark: https://wiki.wireshark.org/TCP_Checksum_Verification .

**Comment by Pusnow** [comment]
I'm not sure if this relates to you, but now you can increase the size of internal buffer by using `setSize`.
You can `git pull` or re-download KENS from Github to use this new feature.
